src/widgets/JudgePanel.ts

```diff
@@ -400,27 +399,44 @@ export class JudgePanel extends BoxPanel {
     };
 
     if (restartKernel) {
-      await kernel.restart();
+      await session.restartKernel();
     }
 
-    // TODO 최대 대기 시간 정의
     const waitIdleState = new Promise<void>((resolve, reject) => {
       const resolveOnIdleState = (
-        sender: IKernelConnection,
+        sender: ISessionContext,
         state: KernelMessage.Status
       ) => {
         if (state === 'idle') {
-          kernel.statusChanged.disconnect(resolveOnIdleState);
+          session.statusChanged.disconnect(resolveOnIdleState);
           resolve();
         }
       };
-      if (kernel.status === 'idle') {
+      if (session.kernelDisplayStatus === 'idle') {
         resolve();
       } else {
-        kernel.statusChanged.connect(resolveOnIdleState);
+        session.statusChanged.connect(resolveOnIdleState);
       }
     });
```

Comment:
This might not be necessary if we using `session.restartKernel`. Could you test it?